test/evp_libctx_test.c

```diff
@@ -473,9 +473,15 @@ static int name_cmp(const char * const *a, const char * const *b)
 
 static void collect_cipher_names(EVP_CIPHER *cipher, void *cipher_names_list)
 {
-    STACK_OF(OPENSSL_CSTRING) *names = cipher_names_list;
-
-    sk_OPENSSL_CSTRING_push(names, EVP_CIPHER_name(cipher));
+    STACK_OF(OPENSSL_STRING) *names = cipher_names_list;
+    const char *name = EVP_CIPHER_name(cipher);
+    char *namedup = NULL;
+
+    /* the cipher will be freed after returning, strdup is needed */
+    if (name != NULL
```

Comment:
Do we allow it to be NULL?

Reply:
I do not think it can be NULL apart from an internal error of some kind.

Reply:
Yeah, so why check for that, rather than assume that it's going to be non-NULL?  A crash of this test at this point would be a strong indicator that the internals got screwed up, especially since we have no other choice (because this function has the void return type) to indicate that something's wrong...

Reply:
OK, I'll add an assert(name != NULL) instead.

Reply:
Fixup pushed